compaction/compaction_manager.cc

```diff
@@ -1454,6 +1454,36 @@ future<compaction_manager::compaction_stats_opt> compaction_manager::perform_tas
     co_return co_await perform_task(seastar::make_shared<TaskType>(*this, &t, std::move(options), std::move(owned_ranges_ptr), std::move(sstables), std::move(compacting), std::forward<Args>(args)...));
 }
 
+template<typename TaskType, typename... Args>
+requires std::derived_from<TaskType, compaction_task_executor> &&
+         std::derived_from<TaskType, compaction_task_impl>
+future<compaction_manager::compaction_stats_opt> compaction_manager::perform_task_on_all_files(std::optional<tasks::task_info> info, table_state& t, sstables::compaction_type_options options, owned_ranges_ptr owned_ranges_ptr, get_candidates_func get_func, Args... args) {
```

Comment:
patch change log can mention that the added `perform_task_on_all_files` version is aimed at replacing the existing one once the conversions in the series are all done.